src/jobsautoreport/query.py

```diff
@@ -99,6 +99,13 @@ def query_average_duration_of_jobs(
 
     def _query_and_log(self, query: dict) -> list[JobDetails]:
         logger.info("OpenSearch query: %s", query)
-        res = self._os_client.search(body=query, index=self._index)
-        logger.info("OpenSearch response: %s", res)
-        return utils.parse_jobs(res)
+        jobs = self._scan(query)
+        return utils.parse_jobs(jobs)
+
+    def _scan(self, query: dict):
```

Comment:
```suggestion
    def _scan(self, query: dict[str, Any]) -> list[dict[Any, Any]]:
```
It's good to fully annotate the types